planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp

```diff
@@ -390,11 +390,17 @@ BehaviorModuleOutput PullOverModule::plan()
 
   // Check if it needs to decide path
   if (status_.is_safe) {
+    const auto ego_segment_idx = motion_utils::findNearestSegmentIndex(
+      getCurrentPath().points, current_pose, std::numeric_limits<double>::max(), M_PI_2);
```

Comment:
Any reason not to use `findFirstNearestSegmentIndexWithSoftConstraints`?

Reply:
I did not intend to change the behavior of the original code.
Based on this idea,
should we add 
```
    if (ego_segment_idx = boost::none) {
      break;
    }
```
after this line?
Otherweise, `ego_segment_idx` can be 0 by using `findFirstNearestSegmentIndexWithSoftConstraints`?

Reply:
I got it. Now I agree with your idea not to change the behavior of the original code.